src/iuse.cpp

```diff
@@ -3366,6 +3376,9 @@ int iuse::solder_weld( player *p, item *it, bool, const tripoint& )
             }
             fix.damage = 0;
         }
+    } else if (fix.damage == 0 || fix.has_flag("PRIMITIVE_RANGED_WEAPON")) {
```

Comment:
This `||` looks suspicious. It means any item that has 0 damage will match the condition, which means one can not reinforce *any* item at all. Ever. Is that really intended? Extra peculiar because the other two places that have similar logic use `&&`, which makes more sense.